Source/ImagePickerController.swift

```diff
@@ -386,7 +386,9 @@ extension ImagePickerController: CameraViewDelegate {
     galleryView.fetchPhotos() {
       guard let asset = self.galleryView.assets.first else { return }
       if self.configuration.allowMultiplePhotoSelection == false {
-        self.stack.assets.removeAll()
+        for asset in self.stack.assets {
+          self.stack.dropAsset(asset)
```

Comment:
A couple of thoughts on this: `ImagePicker` has 2 levels of notifications: 1st: Push/Drop and 2nd: Reload. It was made intentionally to disambiguate between iterative state changes (i.e. when user adds or removes single image) and complete state changes (i.e. when user replaces set of images completely with another set - stack is being reloaded).

In line `self.stack.assets.removeAll()` there is a leaking abstraction, as the `ImagePickerController` uses internal implementation of `ImageStack` and hence this action does not emit notification.

To fix it I'd recommend replace that line with `self.stack.resetAssets([])` - this will emit the notification and won't expose any internal implementation of `ImageStack`.

What's wrong with iteratively removing every photo? The client may render different animations depending on whether you remove one photo or reload stack completely, therefore, it's better to keep this logic intact.